tests/jax/components/executing/observing_test.py

```diff
@@ -0,0 +1,340 @@
+# python3
+# Copyright 2021 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for FeedforwardExecutorObserve class for Jax-based Mava systems"""
+
+
+from types import SimpleNamespace
+from typing import Dict, Tuple
+
+import jax.numpy as jnp
+import pytest
+from acme.types import NestedArray
+from dm_env import StepType, TimeStep
+
+from mava.components.jax.executing.observing import (
+    ExecutorObserveConfig,
+    FeedforwardExecutorObserve,
+)
+from mava.systems.jax.executor import Executor
+from mava.types import OLT
+from mava.utils.sort_utils import sort_str_num
+
+
+class MockAdder:
+    def __init__(self) -> None:
+        pass
+
+    def add_first(
+        self, timestep: TimeStep, extras: Dict[str, NestedArray] = ...  # type: ignore
```

Comment:
is the `= ...` needed? Removing it might sort out the type issue?

Reply:
I wanted to let the same format of the original method that's why. Should I keep it or not?

Reply:
I removed "=..." to get mypy functional

Reply:
I think then keep it in.

Reply:
I think the base adders just pass in an empty dict for the extras. Perhaps you can do that?